### PR TITLE
Adds Pancakes, Pancake Stacking

### PR DESCRIPTION
**Pull request by TehZombehz**
pancakeexample

Adds regular, blueberry, and chocolate chip pancakes as well as stacking
mechanics. Stacking mechanics include:
- Visual pancake stacking and removal as the pancakes added or are eaten
- Progressive pancake stack consumption, IE eating the top pancake first
- Pancake stacks consisting of any variety of pancakes
- Combining stacks of pancakes

I'll add the ability to apply syrup and butter in a later PR, should this work out.

*Please help me clean up this code, if necessary. It might be a tad messy, not unlike sticky pancakes.*

:cl:
add: Nanotrasen Culinary Division has authorized the construction of pancakes, including blueberry and chocolate chip pancakes. Pancakes can be stacked on top of each other.
/:cl:

Why: I felt like the presence of waffles also called for the presence of pancakes. I wanted pancakes to be more realistic and interesting in the ability to stack them, as well. They are part of a complete breakfast!

### CODE REVIEW

**Comment by CosmicScientist** [comment]
Muh culturally French pancakes are being oppressed by these evil Scottish ones.

**Comment by GunHog** [comment]
No syrup? No buttered versions?

**Comment by RemieRichards** [comment]
Real pancakes are had with sugar and orange juice, not syrup and butter

**Comment by GunHog** [comment]
_Takes the bait_

These are American pancakes, as your versions of them are actually _flat!_ These are described as "fluffy". Therefore, UK tastes cannot be applied to these, except in such a case where UK tastes allow or demand sugar and orange juice with American pancakes as well.

**Comment by RemieRichards** [comment]
I am talking about real pancakes gunhog, real pancakes.

**Comment by GunHog** [comment]
I am having too much fun with you, RR!

These ARE real pancakes, RR! Your reality is generated by a Tranquility Lounger! In your strange word, pancakes are **flat and thin!** Yucky! American pancakes are the real ones! _chuckle_

**Comment by RemieRichards** [comment]
lies can never be truth, gunhog, accept your fate.

**Comment by Arianya** [comment]
I hope theres no cap on stacking them

**Comment by GunHog** [comment]
Lies are accepted as truth all the time, RR! Such lies include pancakes being flat!

**Comment by CosmicScientist** [comment]
>orange juice

Hands off them! You'll get nought but lemon juice!

Also if maple syrup gets in I'm going to have to add golden syrup.

Also also I'm going to have to bring in flat pancakes and Dutch pancakes to combat these Welshy Scotch abominations. To culinary war! Before Shrove Tuesday or never!

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 1]
```diff
@@ -350,3 +350,111 @@
 	filling_color = "#F2CE91"
 	tastes = list("pastry" = 1, "sweetness" = 1)
 	foodtype = GRAIN
+
+#define PANCAKE_MAX_STACK 10
```
Comment (lzimann) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: add an `#undef PANCAKE_MAX_STACK` at the end of this file

**Comment by FantasticFwoosh** [comment]
There are two kind of pancakes, these look like thick mini pancakes that are served with butter and come ready-made in a packet (in the UK) but overall it looks good anyway and its nice to see it come into fruition after our chat on IRC.

**Comment by Iamgoofball** [comment]
@TehZombehz 
Can you greyscale these? I'll roll these into goofkitchen.

**Comment by Iamgoofball** [comment]
Also, the "packet" pancakes are known as Crepes.

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm [hunk 2]
```diff
@@ -90,6 +90,33 @@
 	result = /obj/item/reagent_containers/food/snacks/rofflewaffles
 	subcategory = CAT_PASTRY
 
+/datum/crafting_recipe/food/pancakes
+	name = "Pancake"
+	reqs = list(
+		/obj/item/reagent_containers/food/snacks/pastrybase = 1
+	)
+	result = /obj/item/reagent_containers/food/snacks/pancakes
+	subcategory = CAT_PASTRY
```
Comment (kevinz000) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm]: cat made pastries 😏

**Comment by Supermichael777** [comment]
American pancakes are far closer to an en english muffin with the interior structure well suited to liquid spreads
loses out if you flip them too soon and end up with a hard flat on both sides so your syrup runs of the pancakes so they all end up dry except for the soggy one at the bottom.

**Comment by WJohn** [comment]
The blueberry pancakes kind of look like they're covered with bugs.

Though real ones also look that way. Still, would be better if they were less contrasting and a bit more blurry

**Comment by DoctoreMad** [comment]
@WJohn I don't see it so it might be just you

**Comment by CosmicScientist** [comment]
The blueberry muffin sprite might be good to bring up as a comparison.

**Pull request merged**